cpp/src/parquet/file_reader.h

```diff
@@ -58,6 +59,11 @@ class PARQUET_EXPORT RowGroupReader {
   // column. Ownership is shared with the RowGroupReader.
   std::shared_ptr<ColumnReader> Column(int i);
 
+  // Construct a RecordReader for the indicated row group-relative column i.
+  // Ownership is shared with the RowGroupReader.
+  std::shared_ptr<internal::RecordReader> RecordReader(
```

Comment:
Emm I'm ok with the logic here, but is it ok to expose `internal::` logic?

Reply:
I doubt tests checking exposure of internal namespace will complain about this.

Reply:
I'm fine with moving it out of the internal namespace.

Reply:
Sounds good. I think it is better to remove the internal namespace in a separate pull request.

Reply:
Hi @fatemehp . Should we first remove internal, then work on this patch? Or you'd like to let us check this first?

After checking I think maybe we can remove `internal`. cc @pitrou @wgtmac

Reply:
I would prefer to check this in first.

Reply:
IMHO, I'd prefer to move RecordReader out of internal namespace first. Because we can know the consequence of namespace change before exposing it.